docs/writing-intel-modules.md

```diff
@@ -0,0 +1,199 @@
+    identifies the object.  Therefore, we should access an object's ARN using `data['arn']` as opposed to
+    using `data.get('arn')` (the former will raise a `KeyError` if `arn` does not exist and the latter will just return
+    `None` without an exception).
+
+    We _want_ the sync to fail if an important field is not present in our data.  The idea here is that
+    it is better to fail a sync than to add malformed data.
+
+    On the other hand, we should use `data.get('SomeField')` if `SomeField` is something optional that can afford to be `None`.
+
+
+### Load
+
+The `load` function [ingests the processed data to neo4j](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/compute.py#L442).
+
+- The cartography `update_tag`:
+
+    `cartography`'s global [config object carries around an `update_tag` property](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/cli.py#L91-L98) which is
+    [set to the time that the CLI is run](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/sync.py#L131-L134).
+    All `cartography` intel modules need to set the `lastupdated` property on all nodes and all relationships to this
+    `update_tag`.  You can see a couple examples of this in our
+    [AWS ingestion code](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/aws/__init__.py#L106) and our
+    [GCP ingestion code](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/__init__.py#L134).
+
+
+- At minimum, set these fields on all nodes:
+    - `id`
+        Set this to a field that uniquely identifies the node.  For our example of GCP VPCs, we will use the `partial_uri`
+        as an unique identifier for `id` - this is shown in our
+        [schema docs](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/docs/schema/gcp.md#gcpvpc).
+        For an AWS node, it would make sense to set `id` to an ARN.
+
+        If possible, we should use API-provided fields for IDs and not create derived IDs based on multiple fields.
+        In some cases though this is unavoidable -
+        see [GCPNetworkTag](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/docs/schema/gcp.md#gcpnetworktag).
+
+        When setting an `id`, ensure that you also include the field name that it came from.  For example, since we've
+        decided to use `partial_uri`s as GCPVpc `id`,  we should include both `partial_uri` _and_ `id` on the node.  
+        This way, a user can tell what fields were used to derive the `id`.  This is accomplished [here](https://github.com/lyft/cartography/blob/8d60311a10156cd8aa16de7e1fe3e109cc3eca0f/cartography/intel/gcp/compute.py#L455-L457).
+
+    - `lastupdated` - See the special section below on how to set this.
```

Comment:
We should link to the special section.  Github markdown allows for direct linking.